packages/terra-overlay/src/Overlay.jsx

```diff
@@ -0,0 +1,171 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import FocusTrap from 'focus-trap-react';
+import 'terra-base/lib/baseStyles';
+import './Overlay.scss';
+
+const propTypes = {
+  /**
+  * The content to be displayed within the overlay.
+  */
+  children: PropTypes.node,
+  /**
+  * Incidates if the overlay is open.
+  */
+  isOpen: PropTypes.bool,
+  /**
+  * The visual theme to be applied to the overlay background. Accepts 'light', 'dark', and 'clear'.
+  */
+  backgroundStyle: PropTypes.oneOf(['light', 'dark', 'clear']),
+  /**
+  * Incidates if the overlay content is scrollable.
+  */
+  isScrollable: PropTypes.bool,
+  /**
+  * Callback triggered on overlay click or ESC key. Setting this enables close behavior.
+  */
+  onRequestClose: PropTypes.func,
+  /**
+  * Indicates if the overlay is relative to the triggering container.
+  */
+  isRelativeToContainer: PropTypes.bool,
+};
+
+const KEYCODES = {
```

Comment:
It might be something to add to the war room agenda. I agree it would be nice to have it in one location; it would help keep things consistent.